### PR TITLE
[Doc] Add doc for init sub config in DLQ policy and producer

### PR DESCRIPTION
**Pull request by RobertIndie**
### Motivation

This PR is the doc for #13355

### Modifications

* Add doc for init sub config in DLQ policy and producer

### Documentation

- [x] `doc`

### CODE REVIEW

File: site2/docs/concepts-messaging.md [hunk 1]
```diff
@@ -302,6 +302,22 @@ Consumer<byte[]> consumer = pulsarClient.newConsumer(Schema.BYTES)
                 
 ```
 
+When creating a DLQ topic, there is no subscription by default, so if you don't subscribe to this topic in time, you may lose some messages in the DLQ. You can specify the `initSubscriptionName` to make it automatically create the initial subscription on the DLQ.
```
Comment (momo-jun) [inline site2/docs/concepts-messaging.md:305]: By default, there is no subscription during a DLQ topic creation. Without a just-in-time subscription to the DLQ topic, you may lose messages. To automatically create the initial subscription for the DLQ, you can specify the `initSubscriptionName` parameter.

Comment (momo-jun) [reply site2/docs/concepts-messaging.md:305]:
```suggestion
When creating a DLQ topic, there is no subscription by default, so if you don't subscribe to this topic in time, you may lose some messages in the DLQ. You can specify the `initSubscriptionName` to make it automatically create the initial subscription on the DLQ.
```

Comment (momo-jun) [inline site2/docs/concepts-messaging.md:305]:
```suggestion
By default, there is no subscription during a DLQ topic creation. Without a just-in-time subscription to the DLQ topic, you may lose messages. To automatically create the initial subscription for the DLQ, you can specify the initSubscriptionName parameter.
```

**Comment by momo-jun** [comment]
@Anonymitaet Can you pls take a look at the doc PR？

**Review by codelipenghui — APPROVED** [review]

File: site2/docs/client-libraries-java.md [hunk 2]
```diff
@@ -208,6 +208,7 @@ Name| Type |  <div style="width:300px">Description</div>|  Default
 `batchingMaxMessages` |int|The maximum number of messages permitted in a batch.|1000
 `batchingEnabled`| boolean|Enable batching of messages. |true
 `compressionType`|CompressionType|Message data compression type used by a producer. <br />Available options:<li>[`LZ4`](https://github.com/lz4/lz4)</li><li>[`ZLIB`](https://zlib.net/)<br /><li>[`ZSTD`](https://facebook.github.io/zstd/)</li><li>[`SNAPPY`](https://google.github.io/snappy/)</li>| No compression
+`initialSubscriptionName`|string|Use this config to automatically create an initial subscription when creating the topic. If this field is not set, the initial subscription will not be created.|null
```
Comment (Anonymitaet) [inline site2/docs/client-libraries-java.md:211]:
```suggestion
`initialSubscriptionName`|string|Use this configuration to automatically create an initial subscription when creating a topic. If this field is not set, the initial subscription is not created.|null
```

Write in the simple present tense as much as possible if you are covering facts that were, are, and forever shall be true. Details: https://docs.google.com/document/d/1lc5j4RtuLIzlEYCBo97AC8-U_3Erzs_lxpkDuseU0n4/edit#bookmark=id.e8uqh1awkcnp

File: site2/docs/concepts-messaging.md [hunk 3]
```diff
@@ -302,6 +302,22 @@ Consumer<byte[]> consumer = pulsarClient.newConsumer(Schema.BYTES)
                 
 ```
 
+By default, there is no subscription during a DLQ topic creation. Without a just-in-time subscription to the DLQ topic, you may lose messages. To automatically create the initial subscription for the DLQ, you can specify the initialSubscriptionName parameter.
```
Comment (Anonymitaet) [inline site2/docs/concepts-messaging.md:305]:
```suggestion
By default, there is no subscription during a DLQ topic creation. Without a just-in-time subscription to the DLQ topic, you may lose messages. To automatically create an initial subscription for the DLQ, you can specify the `initialSubscriptionName` parameter.
```

File: site2/docs/client-libraries-java.md (hunk 2 above)
Comment (RobertIndie) [reply site2/docs/client-libraries-java.md:211]: Thanks for your suggestions.

**Pull request merged**